File: Reference Project/AI-SDK-Android/deviceAccess_gatewayProxy/src/main/java/com/cmdc/ai/assist/http/ApiConfig.kt
package com.cmdc.ai.assist.http

object ApiConfig {

    /*const val GATEWAY_BASE_URL = "https://z5f3vhk2.cxzfdm.com:30101"*/

    /*const val TERMINAL_INTELLIGENT_SERVICE_PLATFORM_BASE_URL =
        "https://62b98tux.cxzfdm.com:30101"*/

    const val TERMINAL_INTELLIGENT_SERVICE_PLATFORM_BASE_URL_TEST =
        "https://ivs.chinamobiledevice.com:11443/ai-admin-beta"
    const val TERMINAL_INTELLIGENT_SERVICE_PLATFORM_BASE_URL =
        "https://ivs.chinamobiledevice.com:11443"
    const val BASE_URL = "https://aqua-digital.aipaas.com"

    /*const val WSS_WEBSOCKET_ASR_BASE_URL = "wss://ivs.chinamobiledevice.com:11443/app-ws/v1"*/
    const val WSS_WEBSOCKET_ASR_BASE_URL = "wss://ivs.chinamobiledevice.com:11443"

    const val TIMEOUT: Long = 15000L

    var useAgent: Boolean = false
    var agentBaseUrl = ""
    var apiToken = ""
    var auth_token = ""

    const val ASR_INTELLIGENT_DIALOGUE_API_TEST =
        "wss://ivs.chinamobiledevice.com:11443/ai-admin-beta/app-ws/v2/asr"
    const val ASR_INTELLIGENT_DIALOGUE_API = "$WSS_WEBSOCKET_ASR_BASE_URL/app-ws/v2/asr"
    const val AUTOMATIC_SPEECH_RECOGNITION_API = "$WSS_WEBSOCKET_ASR_BASE_URL/app-ws/v1/asr"
    const val AUTOMATIC_SPEECH_RECOGNITION_PERSISTENT_API =
        "$WSS_WEBSOCKET_ASR_BASE_URL/app-ws/v1/long-asr"
    const val ASR_TRANSLATION_API = "$WSS_WEBSOCKET_ASR_BASE_URL/app-ws/v1/realtime_speech_trans"
    const val GATEWAY_API = "/apgp/pl"
    const val AVAILABLE_MODELS_API = "/v1/models"
    const val MODEL_CHAT_API = "/llm/v1/chat/completions"
    const val OBTAIN_DEVICE_INFORMATION_API = "/v2/customer/device/secret/info"
    const val DEVICE_DATA_REPORT_API = "/v2/customer/device/report"
    const val CHAT_COMPLETIONS_API = "/device-api/ai/v1/chat/completions"

    const val CHAT_BOT_COMPLETIONS_API = "/device-api/ai/v2/chat/completions"
    const val TEXT_TO_AUDIO_API = "/device-api/ai/v2/text-to-audio"
    const val INSIDE_RC_CHAT_API = "/device-api/ai/v1/rc-chat"
    const val TEXT_TO_IMAGE_API = "/device-api/ai/v1/sync-draw"
    const val IMAGE_CLASSIFY_API = "/device-api/ai/v1/image-classify"
    const val SPEECH_FILE_TRANSFER_API = "/device-api/ai/v1/speech-file-transfer"
    const val SPEECH_FILE_TRANSFER_QUERY_API = "/device-api/ai/v1/speech-file-transfer-query"
    const val GET_UPLOAD_AUDIO_URL_API = "/device-api/ai/v1/get-upload-audio-url"
    const val IMAGE_TRANSLATE_API = "/device-api/ai/v1/image-translate"
    const val OCR_API = "/device-api/ai/v1/ocr"
    const val WORD_QUERY_API = "/device-api/ai/v1/word-query"
    const val NOTE_SUMMARY_API = "/device-api/ai/v1/note-summary"
    const val TEXT_TRANSLATE_API = "/device-api/ai/v1/text-translate"

    const val TEXT_TRANSLATE_ALI_YUN_API = "/device-api/ai/v2/text-translate"

}

File: components/ai_sdk/src/api_config.cc
#include "ai_sdk/api_config.h"

namespace ai_sdk {

// ===== 基础URL配置 =====
// 这些配置在编译时确定，提供云端服务的基础信息

/**
 * 终端智能服务平台基础URL
 * 中国移动物联网平台，提供设备管理和通信服务
 * 使用HTTPS协议，端口11443
 */
const char* ApiConfig::TERMINAL_INTELLIGENT_SERVICE_PLATFORM_BASE_URL =
    "https://ivs.chinamobiledevice.com:11443";

/**
 * 网关API路径
 * 用于获取代理服务器配置
 * 完整URL: {BASE_URL}/apgp/pl
 */
const char* ApiConfig::GATEWAY_API = "/apgp/pl";

/**
 * 设备信息获取API路径
 * 设备注册认证接口，通过设备号获取设备凭证
 * 完整URL: {BASE_URL}/v2/customer/device/secret/info
 */
const char* ApiConfig::OBTAIN_DEVICE_INFORMATION_API = "/v2/customer/device/secret/info";

/**
 * 设备数据上报API路径
 * 心跳/数据上报接口，报告设备状态和活跃信息
 * 完整URL: {BASE_URL}/v2/customer/device/report
 */
const char* ApiConfig::DEVICE_DATA_REPORT_API = "/v2/customer/device/report";

/**
 * HTTP请求超时时间（毫秒）
 * 设置为15秒，考虑网络环境和设备性能
 * 可根据实际需求调整：
 * - 降低：提高响应速度，但可能增加超时错误
 * - 提高：减少超时错误，但会增加等待时间
 */
const long ApiConfig::TIMEOUT = 15000L;

// ===== 运行时配置 =====
// 这些配置在程序运行时动态设置

/**
 * 是否使用代理（默认：false）
 * 由GatewayClient在启动时从云端获取代理配置后设置
 * true: 后续请求通过代理服务器转发
 * false: 直接连接云端服务器
 */
bool ApiConfig::useAgent = false;

/**
 * 代理服务器基础URL（默认：空）
 * 当useAgent为true时使用此地址
 * 格式: http(s)://proxy.example.com:port
 */
std::string ApiConfig::agentBaseUrl = "";

/**
 * API访问令牌（默认：空）
 * 用于代理服务器认证
 * 由GatewayClient获取并设置
 */
std::string ApiConfig::apiToken = "";

/**
 * 设备认证令牌（默认：空）
 * 设备注册后获得的认证信息
 * TODO: 目前未使用，需要实现设备签名逻辑
 * 计划用途：为每个API请求生成签名，防止伪造请求
 */
std::string ApiConfig::auth_token = "";

} // namespace ai_sdk